staging/src/k8s.io/legacy-cloud-providers/azure/retry/azure_retry_test.go

```diff
@@ -80,6 +80,7 @@ func TestJitterWithNegativeMaxFactor(t *testing.T) {
 	// jitter := duration + time.Duration(rand.Float64()*maxFactor*float64(duration))
 	// If maxFactor is 0.0 or less than 0.0, a suggested default value will be chosen.
 	// rand.Float64() returns, as a float64, a pseudo-random number in [0.0,1.0).
+	rand.Seed(1)
```

Comment:
I still don't understand why this is necessary... the test is tolerating a range of values, so *any* value chosen by rand() should produce a value that passes the test, right? are the two assertions incorrect?

Reply:
can you add logging to log `expected, res, expected-res` to see what the actual values produced are and why we are failing?

Reply:
it looks like the test is just wrong...

the first call to jitter passes in a negative maxFactor, which means jitter uses a maxFactor of 1, so `res` is between `duration` and `2*duration`.

the second call to jitter passes in maxFactor of 1, so `expected` is also `duration` and `2*duration`.

It is valid for `expected` to be smaller than `res`, so the first assertion is not correct.

It is valid for `expected` to be exactly 2*duration and `res` to be exactly `1*duration`, so the second assertion is also not correct.

Reply:
that exposed the bug in the test, but I think the right fix is to correct the test, not re-seed to mask the test issue

Reply:
Got it, I will update the test accordingly

Reply:
I have updated the assertion to check
1. Jitter value with negative max factor
2. Jitter value with positive max factor